lib/routes/sms.js

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const error = require('../error')
+const libpn = require('google-libphonenumber')
+const validators = require('./validators')
+
+const METRICS_CONTEXT_SCHEMA = require('../metrics/context').schema
+const SENDER_IDS = new Map(require('../sms-sender-ids.json'))
+
+module.exports = (log, isA, error, customs, sms) => {
+  return [
+    {
+      method: 'POST',
+      path: '/sms',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            phoneNumber: isA.string().regex(validators.E164_NUMBER).required(),
+            messageId: isA.number().positive().required(),
+            metricsContext: METRICS_CONTEXT_SCHEMA
+          }
+        }
+      },
+      handler (request, reply) {
+        log.begin('sms.send', request)
+        request.validateMetricsContext()
+
+        const sessionToken = request.auth.credentials
+        const phoneNumber = request.payload.phoneNumber
+        const messageId = request.payload.messageId
+        const acceptLanguage = request.app.acceptLanguage
+
+        customs.check(request, sessionToken.email, 'connectDeviceSms')
```

Comment:
Actually @vbudhram, now I've had a chance to dig into this, I've changed my mind.

The interface for `customs.check` is pretty well established and I don't really want to change it in this PR. I could pass in a modified request object that matches what you expect but that seems like unnecessary complexity. Or I could add an extra `phoneNumber` arg to the end of the signature with a default value, but that seems like special snowflake behaviour for a single case. Neither particularly floats my boat.

Also, in terms of making it easier for people trace through the flow by eye, it seems like we should use the same names everywhere. So `phoneNumber` would be a better choice than `smsNumber` in the customs server code, to match what was agreed in the auth server API.

What do you reckon?